packages/@uppy/utils/src/findDOMElement.ts

```diff
@@ -1,10 +1,10 @@
 import isDOMElement from './isDOMElement.ts'
 
-export default function findDOMElement<T>(
-  element: T,
+export default function findDOMElement<E, ReturnType = Element>(
+  element: E,
   context: Document = document,
-): T extends Element ? T
-: T extends Node | string ? Element | null
+): ReturnType extends Element ? ReturnType
+: ReturnType extends Node | string ? Element | null
```

Comment:
I'm not sure if we need the added complexity, wouldn't we be better of by using `as HTMLFormElement` rather than adding another generic?

Reply:
The problem this mostly solves, is that what we pass is in is not the same type as what we get back. That was the case before. But if I pass in `this.opts.target`, that shouldn't be my return value.

Reply:
Maybe we'd get better results with `T extends Element ? T & Element : …`? I'm -1 on adding a second generic, I'd rather have us use assertions

Reply:
FWIW I tested with `T&Element`, and that doesn't work. But anyway, I think we should use an assertion, either a TS one (so `as HTMLFormElement`), or a runtime one:

```js
function assertHTMLFormElement(input: unknown): HTMLFormElement {
  if ((input as Node)?.nodeName !== 'FORM') {
    throw new Error('ASSERTION FAILED: the target is not a <form> element', {cause: input})
  }
  return input as any
}
```

That might be a case where a runtime check can be more helpful for debugging than a TypeError at some random place.